botometerAnalyser/routes.js

```diff
@@ -44,6 +36,54 @@ Roger! I'm analysing the probability that the accounts that have tweeted **"${se
 ${activeJobsCount ? '\n:mantelpiece_clock: _I’m already running an analysis, I will search for yours as soon as possible._' : '\n_In order to limit that search to 30 minutes, we will stop at 100 accounts max._'}
 `
 	});
+}
+
+async function startRetweetPipeline({ req, res, screenName, tweetId, tweetUrl }) {
+	const activeJobsCount = await botometerQueue.getActiveCount();
+
+	retweets.analyse({
+		screenName,
+		tweetId,
+		tweetUrl,
+		responseUrl: req.query.response_url,
+		requesterUsername: req.query.user_name
+	});
+
+	res.json({
+		response_type: 'in_channel',
+		text: `
+Roger! I'm analysing the probability that the accounts that have retweeted that tweet by @${screenName} are robots.
+${activeJobsCount ? '\n:mantelpiece_clock: _I’m already running an analysis, I will search for yours as soon as possible._' : '\n_In order to limit that search to 25 minutes, we might not analyse all retweeters._'}
+`
+	});
+}
+
+
+router.get('/', async (req, res, next) => {
+	const { token: givenToken, text } = req.query;
+
+	if (givenToken !== mattermostToken) {
+		return res.status(401).json({ Error: 'Missing or invalid token' });
+	}
+
+	if (!text) {
+		return res.json({
+			text: 'Hey! I can help you by analyzing the latest shares on a specific topic but I need a query.\nSo, give me a keyword, an URL or some text. For example: \n`/botometer disinformation`'
+		});
+	}
+
+	const retweetRegexp = /^https:\/\/twitter\.com\/([^\/]*)\/status\/(\d+)$/;
+	const retweetRegexpResult = text.match(retweetRegexp);
+	if (retweetRegexpResult) {
+		const screenName = retweetRegexpResult[1];
```

Comment:
FYI: you could have used ES6 destructuring like this: `[, screenName, tweetId ] = retweetRegexpResult`

Reply:
:+1: